examples/step-68/step-68.cc

```diff
@@ -0,0 +1,810 @@
+#include <deal.II/particles/generators.h>
+
+// Since the particles do not form a triangulation, they have their
+// own specific DataOut class which will enable us to write them
+// to commonly used parallel vtu format:
+#include <deal.II/particles/data_out.h>
+
+#include <cmath>
+#include <iostream>
+
+namespace Step68
+{
+  using namespace dealii;
+
+  // @sect3{Run-time parameter handling}
+
+  // Similarly to what is done in step-60, we set up a class that holds
+  // all the parameters of our problem and derive it from the ParameterAcceptor
+  // class to simplify the management and creation of parameter files.
+  //
+  // The ParameterAcceptor paradigm requires all parameters to be writable by
+  // the ParameterAcceptor methods. In order to avoid bugs that would be very
+  // difficult to track down (such as writing things like `time = 0` instead of
+  // `time == 0`), we declare all the parameters in an external class, which is
+  // initialized before the actual `ParticleTracking` class, and pass it to
+  // the main class as a `const` reference.
+  //
+  // The constructor of the class is responsible for the connection between the
+  // members of this class and the corresponding entries in the
+  // ParameterHandler. Thanks to the use of the
+  // ParameterHandler::add_parameter() method, this connection is trivial, but
+  // requires all members of this class to be writable.
+  class ParticleTrackingParameters : public ParameterAcceptor
+  {
+  public:
+    ParticleTrackingParameters();
+
+    // This class consists largely of member variables that
+    // describe the details of the particle tracking simulation and its
+    // discretization. The following parameters are about where output should
+    // land, the spatial discretization of the velocity (the default is $Q_1$),
+    // the time step and the output frequency (how many time steps should
+    // elapse before we generate graphical output again):
+    std::string output_directory = "./";
+
+    unsigned int velocity_degree       = 1;
+    double       time_step             = 0.002;
+    double       final_time            = 4.0;
+    unsigned int output_frequency      = 10;
+    unsigned int repartition_frequency = 5;
+
+    // We allow every grid to be refined independently. In this tutorial, no
+    // physics is resolved on the fluid grid, and its velocity is calculated
+    // analytically.
+    unsigned int fluid_refinement              = 4;
+    unsigned int particle_insertion_refinement = 3;
+  };
+
+  // There remains the task of declaring what run-time parameters we can accept
+  // in input files. Since we have a very limited number of parameters, all
+  // parameters are declared in the same section.
+  ParticleTrackingParameters::ParticleTrackingParameters()
+    : ParameterAcceptor("Particle Tracking Problem/")
+  {
+    add_parameter(
+      "Velocity degree", velocity_degree, "", this->prm, Patterns::Integer(1));
```

Comment:
Why is it that this one call is different from the others in that it gets `this->prm` and `Patterns::Integer(1)` as arguments? Is the `prm` a default argument and you need to specify it because you want to provide a pattern? I think that some of the other arguments could also benefit from a pattern.

Reply:
Indeed, that was a clear mistake on my part. I fixed it and added patterns for the doubles and integer as well as a min when it is pertinent.